### DISCUSSION TITLE
Exercise Catalog as the Home Page and App Bar

### DISCUSSION THREAD

**Discussion by eliasfaltin** [GECO-Project-App/facilitation-app#57] [4 - Shipped]
# Hypothesis
Apps need a home screen and main menu/menu bar. Duh
But what should be on our Home Screen? What other screens should be linked from the menu bar?
Here is a list of ideas:
- An account-based counter of the value we created. Instacart, for example, shows you how much time you have saved on not having to do grocery shopping while using their app.
- An AI chat where the underlying model is a RAG-based system trained on/ has access to facilitation materials and the catalogue of exercises built into the app. This way, it can both coach facilitators on their unique problems as well as help them build agendas from the exercise catalog
- **The exercise catalog: A categorized list of exercises built into the app.**
- The "Agenda" view shows a list of pre-built or custom-configured agendas

The first step is to display a catalog of available exercises front and center. This also means we won't need to implement a menu bar (quite yet).

The hypothesis is that this enables users to choose a suitable exercise more quickly for their current situation/need. Currently, we only show a simple button with the exercise title, which does not provide sufficient information for inexperienced facilitators
# MVP/ Experiment
- The ability to switch the app's language at the top of the home screen page (whether this is the catalog or a different page in the future).

As a user, I want each item in the catalog to show:
- The name of the exercise
- How long the exercise takes to complete (approximately)
- The group/team size the exercise can support
- A brief description of why the exercise is valuable to do
so that I can make an informed decision on whether a given exercise is the right one for my team right now.

I am unsure which metadata to use to sort exercises in the catalog. Options are:
- In the order they might typically be used in a meeting (Check-in, SSC, Check out)
- Alphabetically (CI, CO, SSC)
- Randomize on page load (this could give us the "cleanest" data on CTR (click-through rate) and completion rate)
# Measure
For now, we can't test this against the existing implementation since we
1. don't have a big user base
2. our Posthog implementation didn't work until earlier today, so we weren't collecting data
I propose that we add events for:
- When a user visits the exercise page
- when a given exercise is selected from the catalog
- when a given exercise is completed

This allows us to track the CTR, popularity (=usefulness?), and "completion percentage" of any given exercise in the catalog.
# Learn
Learning from quantitative data will be limited at this point. Plus, this feature is a "table stakes" feature that the app just needs to have.

I assume we will eventually reach a point where the exercise catalog becomes so full that we need to categorize exercises.

**Comment by alice-ang** [comment]
I don't know if this is relevant in this discussion or should be somewhere else, but should we have a language switcher on the home page? Or should users be able to switch language from anywhere?

**Reply by eliasfaltin** [reply]
I added this point now

**Comment by eliasfaltin** [comment]
![CleanShot 2024-09-23 at 17 40 40@2x](https://github.com/user-attachments/assets/1674f56a-93dd-4b52-8049-b9be6d6dc1fd)
☝️ initial design from @aripuguh 

The Header with the GECO gecko and language switcher stays pinned to the page and the exercise cards scroll "behind it".
